nixos/modules/hardware/video/nvidia.nix

```diff
@@ -226,31 +235,60 @@ in
     environment.systemPackages = [ nvidia_x11.bin nvidia_x11.settings ]
       ++ filter (p: p != null) [ nvidia_x11.persistenced ];
 
+    systemd.packages = optional cfg.powerManagement.enable nvidia_x11.out;
+
+    systemd.services = let
+      baseNvidiaService = state: {
+        description = "NVIDIA system ${state} actions";
+
+        path = with pkgs; [ kbd ];
+        serviceConfig = {
+          Type = "oneshot";
+          ExecStart = "${nvidia_x11.out}/bin/nvidia-sleep.sh '${state}'";
```

Comment:
Broken with new driver due to conditional install (silent failure in package install process from this PR) if the script doesn't exist:

https://github.com/NixOS/nixpkgs/issues/127693